### PR TITLE
Pdata json api

### PR DESCRIPTION
**Pull request by barnabwhy**
Adds a JSON API to get player data.
Would be useful were someone to make a stat tracking web app or something

----
<details>
<summary><h2>Endpoints:</h2></summary>

### /player/pdata

- All pdata (excluding pdiff)
- Pretty much encompasses all the below and more, the other ones are for getting more specific information
- This endpoint notably **excludes** username

### /player/info
-  Username (if #42 gets merged)
-  Player id
-  Gen
-  XP
- Calling card
- Callsign icon
- Net worth

### /player/stats
-  Player id
-  Gen
-  XP
- Credits
- Net worth
- Faction XP
- Titan XP
- FD titan XP
- Game stats
- Map stats
- Time stats
- Distance stats
- Weapon stats
- Weapon kill stats
- Kill stats
- Death stats
- Misc stats
- FD stats
- Titan stats
- KD ratio
- PVP KD ratio
- Win streak
- Highest win streak

### /player/loadout
-  Faction
- Full active pilot loadout
- Full active titan loudout
- Full pilot loadout list
- Full titan loadout list

### CODE REVIEW

**Comment by barnabwhy** [comment]
Closes #31

File: client/playerinfo.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+module.exports = ( fastify, opts, done ) =>
+{
+	// exported routes
+
+	// GET /player/pdata
+	// show pdata for a given player as json
+	fastify.get( "/player/pdata",
+		{
+			config: { rateLimit: getRatelimit( "REQ_PER_MINUTE__REDIRECT" ) }, // ratelimit
```
Comment (abarichello) [inline client/playerinfo.js:18]: Player data should have its own category of ratelimit

File: client/playerinfo.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+			Object.assign( ret, pdataFiltered )
+
+			return ret
+		} )
+
+	// GET /player/loadout
+	// show louadout data for a given player as json
+	fastify.get( "/player/loadout",
+		{
+			config: { rateLimit: getRatelimit( "REQ_PER_MINUTE__REDIRECT" ) }, // ratelimit
```
Comment (abarichello) [inline client/playerinfo.js:116]: replace with the same ratelimit category as the routes above

File: client/playerinfo.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+			}
+			let pdata = PdataToJsonUntyped( account.persistentDataBaseline, PLAYER_DATA_PDEF_231 )
+			return pdata
+		} )
+
+	// GET /player/info
+	// show info for a given player as json
+	fastify.get( "/player/info",
+		{
+			config: { rateLimit: getRatelimit( "REQ_PER_MINUTE__REDIRECT" ) }, // ratelimit
```
Comment (abarichello) [inline client/playerinfo.js:40]: replace with the same ratelimit category as the routes above

File: client/playerinfo.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+			Object.assign( ret, pdataFiltered )
+
+			return ret
+		} )
+
+	// GET /player/stats
+	// show stats for a given player as json
+	fastify.get( "/player/stats",
+		{
+			config: { rateLimit: getRatelimit( "REQ_PER_MINUTE__REDIRECT" ) }, // ratelimit
```
Comment (abarichello) [inline client/playerinfo.js:77]: replace with the same ratelimit category as the routes above

**Review by abarichello — APPROVED** [review]

**Pull request merged**